### ISSUE TITLE
Hibernate validation not happening for @BeanParam with groups

### ISSUE THREAD

**Issue by sureshrengasamy** [dropwizard/dropwizard#1433]
Issue is [fixed](https://github.com/dropwizard/dropwizard/issues/1227) for validating @BeanParam object but it is not working when validator groups are present.

```
public class BaseClass {

    @NotNull(groups=SignupValidator.class)
    @QueryParam("startDate")
    protected LocalDate startDate;

    public LocalDate getStartDate() {
        return startDate;
    }
}

@Path("path")
public class QuantityResource {

    @GET
    public ResponseEntity working(@Validated(SignupValidator.class) @BeanParam BaseClass req) {
        ...    
    }

}
```

**Comment by nickbabcock** [comment]
I'm marking this as a bug, as currently `Validated` only works on request entities, and one shouldn't have to declare a dummy request object just so they can get the `Validated` annotation to work. I'm proposing a fix that will let the `Validated` annotation be on any parameter but if more than one `Validated` annotation is supplied with a different set of groups then it's a runtime error because the underlying hibernate validator is unable to perform validation against parameters with different groups.

**Comment by sureshrengasamy** [comment]
Thank you

**Comment by arteam** [comment]
Closed via #1436
